test/Table_Tests/src/IO/Excel_Spec.enso

```diff
@@ -103,22 +105,42 @@ spec_fmt suite_builder header file read_method sheet_count=5 =
 
         group_builder.specify "should let you read all sheets into a single table" <|
             wb = read_method file
-            action = wb.read_many on_problems=_
-            tester table =
-                table.row_count . should_equal 25
-                table.column_names . should_equal ["Sheet Name", "Name", "Quantity", "Price", "A", "B", "C", "D", "E", "Student Name", "Enrolment Date", "Item", "Price 1"]
-            problems = [Empty_Sheet.Error, Duplicate_Output_Column_Names.Error ["Price"]]
-            Problems.test_problem_handling action problems tester ignore_warning_cardinality=True
+            r1 = wb.read_many on_problems=..Report_Error
+            r1.should_fail_with Empty_Sheet
+
+            r2 = wb.read_many on_problems=..Report_Warning
+            r2.row_count . should_equal 25
+            r2.column_names . should_equal ["Sheet Name", "Name", "Quantity", "Price", "A", "B", "C", "D", "E", "Student Name", "Enrolment Date", "Item", "Price 1"]
+            # We also ensure how many rows were loaded from each sheet. There should be no entries for the "Empty" sheet.
+            r2.at "Sheet Name" . to_vector . should_equal <|
+                (Vector.fill 6 "Simple") + (Vector.fill 7 "Strange Dimensions") + (Vector.fill 6 "Dates") + (Vector.fill 6 "Duplicate Columns")
+            w1 = Problems.expect_warning No_Rows r2
+            w1.to_display_text . should_contain "The sheet Empty failed to load"
+            w1.to_display_text . should_contain "There is no data in the sheet."
+            w2 = Problems.expect_warning Duplicate_Output_Column_Names r2
+            w2.column_names . should_equal ["Price"]
 
         group_builder.specify "should let you read all sheets into a table of tables" <|
             wb = read_method file
-            action = wb.read_many return=..Table_Of_Tables on_problems=_
+            action = wb.read_many return=..With_New_Column on_problems=_
```

Comment:
We should keep a deprecrated test too,

Reply:
There is one just below. I'll add a warning check as well.